Sources/SwiftDocC/Infrastructure/Diagnostics/DiagnosticEngine.swift

```diff
@@ -32,6 +32,9 @@ public final class DiagnosticEngine {
             self.filter = { $0.diagnostic.severity.rawValue <= self.filterLevel.rawValue }
         }
     }
+    
+    /// Determines whether warning will be treated as error
+    private let warningsAsErrors: Bool
```

Comment:
Nit: I would prefer naming this `treatWarningsAsErrors` because `warningsAsErrors` could be understood as a property of type `[Diagnostic]`, i.e., the warnings that have been converted to errors. Same goes for the other instances of "warningsAsErrors"

Reply:
Got it. It was originally be “let treatWarningsAsErrors: Bool” and after we agree to use the name “warnings-as-errors” I replace all the name to align with it.

So I guess the variable name should still use “treatWarningsAsErrors” and the CLI flag option should be “—-warnings-as-errors”.

Reply:
Right, if the property in `Convert.swift` is `warningsAsErrors`, I'm fine with that because of how ArgumentParser works.